### ISSUE TITLE
should.be.a('function') does not accept a callable object created from v8::ObjectTemplate

### ISSUE THREAD

**Issue by swordfeng** [chaijs/chai#698]
I have created objects from `v8::ObjectTemplate` with `SetCallAsFunctionHandler` set.
When I test the objects with `typeof`, it tells me that they are `function`s. And also I can invoke them as functions by calling `obj()`.
But when I test them by `obj.should.be.a('function')`, the assertion fails.

**Comment by swordfeng** [comment]
test environment:
node.js 6.0.0
nan 2.3.3
chai 3.5.0

test.cc:
```
#include <node.h>
#include <nan.h>

void Func(const Nan::FunctionCallbackInfo<v8::Value> &args) {
}
void Init(v8::Local<v8::Object> exports) {
    v8::Local<v8::ObjectTemplate> tpl = Nan::New<v8::ObjectTemplate>();
    Nan::SetCallAsFunctionHandler(tpl, Func);
    v8::Local<v8::Object> obj = tpl->NewInstance();
    exports->Set(Nan::New("obj").ToLocalChecked(), obj);
}

NODE_MODULE(addon, Init)
```
test.js:
```
require('chai').should();
var addon = require('./build/Release/addon.node');
console.log(typeof addon.obj); // outputs 'function'

addon.obj.should.be.a('function'); // throws
```

**Comment by keithamus** [comment]
Hey @swordfeng thanks for the issue.

I think this is because our type detection (at current) uses `[[Class]]` (or as its known these days `@@toStringTag`) via `Object.prototype.toString`. It does not use typeof. This will be "fixed" in Chai `4.0.0` because a typeof check is used as an optimisation step.

For now, if you want to use it as a function, ensure a `Symbol.toStringTag` (or `[[Class]]` - not sure what this is in V8 or nan) is set. 

I'll close this issue, but if you want to continue the discussion, feel free 😄
